java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -1120,6 +1122,24 @@ private boolean hasLocallyPresentOptionalBinding(ResolvedBindings resolvedBindin
               && !getLocalExplicitBindings(keyFactory.unwrapOptional(resolvedBindings.key()).get())
                   .isEmpty();
         }
+
+        /**
+         * Returns {@code true} if {@code resolvedBindings} contains a provision binding
+         * for which there is an explicit present binding in this component that was resolved
+         * previously.
+         */
+        boolean hasLocallyConflictingContributionBinding(ResolvedBindings resolvedBindings) {
+          return resolvedBindings
+                .contributionBindings()
+                .stream()
+                .anyMatch(contributionBinding -> contributionBinding.bindingKind().equals(PROVISION)
+                  && !getLocalExplicitBindings(resolvedBindings.key()).isEmpty()
+                  && parentResolver.isPresent()
```

Comment:
Here's the failure:

https://s3.amazonaws.com/archive.travis-ci.org/jobs/243061395/log.txt?X-Amz-Expires=29&X-Amz-Date=20170620T172916Z&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJRYRXRSVGNKPKO5A/20170620/us-east-1/s3/aws4_request&X-Amz-SignedHeaders=host&X-Amz-Signature=06c31fe732edd873eff14667b464953cff160115ecb5739f7ca06c1f91c2c2b6